UI/app state redo

ohmed opened the pull request:
All existing UI should be same and should have same functionality as before.

avaer commented on src/Header.jsx:
```diff
@@ -1,279 +1,365 @@
+
+                    if ( ! state.openedPanel === 'MagicPanel' && ! ioManager.inputFocused() ) {
+
+                        event.preventDefault();
```
And yes, this is not new code. But since we are rewriting event handling, it's a good time to fix it.

avaer reviewed (commented):
Overall looks good.

Should port the mega map screen to the new open tracking system, since it is designed to be modal and exclusive like the other tabs.

ohmed commented:
@alisaad673 fixed all issues listed above.
can be retested [character btn position is the same, looks better like it is as for me].

ohmed commented:
@alisaad673 fixed all issues listed above.
Only don't know about warnings [first item], I don't get any or don't know the reproduce steps.

ohmed commented:
Also fixed:
1. Enter key issue when pressed before onto 'scene-select' panel btns.
2. 'World' panel btn now visible when 'World' panel is opened.
3. Fixed handling 'm' & 'z' btns in chat.

avaer commented:
Looks good but has conflicts.

avaer commented:
I fixed the conflicts.

However, this PR regresses the UI animations further than previous PRs: https://github.com/webaverse/app/issues/2592

avaer commented:
Even after this supposedly passed tests I'm seeing regressions in the animations. It might need another redo, with a strong emphasis on not regressing.

This PR included many other fixes for previously introduced UI regressions but this PR isn't likely to be merged anymore. Can we break those out?

ohmed commented:
Updated panels open/close animations [+character emotions bar animation].

avaer commented:
Mouse wheel seems completely broken in this branch. It does not scroll the camera at all.

alisaad673 commented:
Just tested it again. Mouse wheel is working fine now.

avaer commented:
This PR is still busted in mouse wheel:

```
Unable to preventDefault inside passive event listener invocation.
```

![image](https://user-images.githubusercontent.com/6926057/159152099-ed2cc675-4067-4f1e-9fb9-06790ec13fdc.png)

avaer commented:
There are also conflicts.

avaer commented:
Bugs are fixed after merge.
